avocado/core/nrunner/runnable.py

```diff
@@ -267,9 +270,16 @@ def get_capabilities_from_runner_command(runner_command, env=None):
         out, _ = process.communicate()
 
         try:
-            return json.loads(out.decode())
+            capabilities = json.loads(out.decode())
         except json.decoder.JSONDecodeError:
-            return {}
+            capabilities = {}
+
+        # lists are not hashable, and here it'd make more sense to have
+        # a command as it'd be seen in a command line anyway
+        cmd = " ".join(runner_command)
+        STANDALONE_EXECUTABLE_CONFIG_USED[cmd] = capabilities.get(
```

Comment:
What about update the `STANDALONE_EXECUTABLE_CONFIG_USED` only when the `cmd` key doesn't exist, because this shouldn't change during runtime.

Reply:
Right, that goes along the point of my previous response (caching/reusing it).